quarkus-integration/quarkus/deployment/src/test/java/ai/timefold/solver/quarkus/rest/TimefoldProcessorHotReloadMultipleSolversTest.java

```diff
@@ -0,0 +1,69 @@
+package ai.timefold.solver.quarkus.rest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import ai.timefold.solver.quarkus.testdata.normal.constraints.TestdataQuarkusConstraintProvider;
+import ai.timefold.solver.quarkus.testdata.normal.domain.TestdataQuarkusEntity;
+import ai.timefold.solver.quarkus.testdata.normal.domain.TestdataQuarkusSolution;
+import ai.timefold.solver.quarkus.testdata.shadowvariable.constraints.TestdataQuarkusShadowVariableConstraintProvider;
+import ai.timefold.solver.quarkus.testdata.shadowvariable.domain.TestdataQuarkusShadowVariableEntity;
+import ai.timefold.solver.quarkus.testdata.shadowvariable.domain.TestdataQuarkusShadowVariableListener;
+import ai.timefold.solver.quarkus.testdata.shadowvariable.domain.TestdataQuarkusShadowVariableSolution;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusDevModeTest;
+import io.restassured.RestAssured;
+
+/**
+ * Do not remove the public modifier from this class. Otherwise, the test will fail with the following:
+ * <p>
+ * Caused by: java.lang.IllegalAccessException: class io.quarkus.test.QuarkusDevModeTest cannot access a member of class
+ * ai.timefold.solver.quarkus.rest.TimefoldProcessorHotReloadMultipleSolversTest with modifiers ""
+ * at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:674)
+ * at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:489)
+ * at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
+ * at io.quarkus.test.QuarkusDevModeTest.createTestInstance(QuarkusDevModeTest.java:222)
+ */
```

Comment:
No need to be so verbose. Enough to say that removing public will cause an exception. If someone decides to try even though they've been warned, they'll find out what exception it is. :-)